crates/flox-rust-sdk/src/models/environment/managed_environment.rs

```diff
@@ -448,6 +473,55 @@ fn gcroots_dir(flox: &Flox, pointer: &ManagedPointer) -> PathBuf {
     flox.cache_dir.join("run").join(pointer.owner.to_string())
 }
 
+impl ManagedEnvironment {
+    #[allow(unused)]
+    pub fn pull(&mut self) -> Result<(), ManagedEnvironmentError> {
+        // Fetch the remote branch into FETCH_HEAD
+        self.floxmeta
```

Comment:
suggestion nonblocking: could you replace the fetch + branch_contains_commit + push2 with a single `fetch remote_branch_name:project_branch`?

Reply:
I'd prefer to tell explicitly if the branches diverged

Reply:
What do you mean explicitly? `fetch` will fail if the branches have diverged unless you put `+` in the refspec

Reply:
fetch will fail also when the url is unreachable for example.
I'd like to inform someone explicitly that their environments diverged.
Showing them a raw git exception seems rather uninformative internal detail..
Unless we start parsing git errors (at least some well known ones) i guess trying to fetch the branch first is cheap and effective enough for now.